src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -356,7 +356,7 @@ ValidatorList::verify (
 
     if (list.isMember("sequence") && list["sequence"].isInt() &&
         list.isMember("expiration") && list["expiration"].isInt() &&
-        list.isMember("validators") && list["validators"].isArray())
+        list.isMember("validators") && list["validators"].isArrayOrNull())
```

Comment:
Under what circumstances could `list["validators"]` be a null object here? Is this possible at all, @wilsonianb?

Also, we seem to use this paradigm `obj.isMember("...") && obj["..."].isSomething()` quite a lot. Is it an improvement on the API to have something like:

    list.isMember("expiration, Json::intValue) &&
    list.isMember("validators", Json::arrayValue))

Reply:
`bool isMember(key, type) const` is not a bad idea.

Reply:
`list["validators"]` can be null if someone is using a faulty/malicious published validator list.

Reply:
I'm reading @wilsonianb's comment as meaning that a null `list["validators"]` would either be an error or malicious.  But it either case it should be rejected.  So I would expect the test here to be only `list["validators"].isArray()`.  But I could easily have misunderstood the comment.

Reply:
Sorry, I should have been clearer.
@scottschurr is correct. It is possible to `list["validators"]` to be null, and if so it should be rejected.
So this should be `list["validators"].isArray()`

Reply:
Thanks, done.  Just pushed.